File: 2021_09_29/C++/2.cpp
#include <iostream>

int digit_sum(int n) {
	return n / 100 + (n / 10) % 10 + n % 10;
}

int main() {
	int ans = 0;
	int D_SUM[1000];
	for (int i = 0; i < 1000; i++) D_SUM[i] = digit_sum(i);
	for (int x1 = 100; x1 < 1000; x1++) {
		if (x1 % D_SUM[x1] == 0)
			ans++;
		for (int x2 = x1 + 1; x2 < 1000; x2++)
			if (x1 % digit_sum(x2) == 0 && x2 % digit_sum(x1) == 0)
				ans += 2;
	}
	std::cout << ans << std::endl;
	return 0;
}

File: 2021_09_29/Pascal/2.pas
var ans : longint;
	x1, x2 : longint;
	D_SUM : array [100..999] of longint;
	i : longint;

function digit_sum(n : longint) : longint;
Begin
	digit_sum := n div 100 + (n div 10) mod 10 + n mod 10;
End;

BEGIN
	ans := 0;
	for i := 100 to 999 do D_SUM[i] := digit_sum(i);
	for x1 := 100 to 999 do
	begin
		if (x1 mod D_SUM[x1] = 0) then
			Inc(ans);
		for x2 := x1+1 to 999 do
			if (x1 mod D_SUM[x2] = 0) and (x2 mod D_SUM[x1] = 0) then
				Inc(ans, 2);
	end;
	Writeln(ans);
END.
